server/src/main/java/org/elasticsearch/health/node/DiskHealthIndicatorService.java

```diff
@@ -160,36 +158,61 @@ public HealthIndicatorResult calculate(boolean explain, HealthInfo healthInfo) {
     }
 
     private String getSymptom(
-        boolean clusterHasBlockedIndex,
         Set<String> blockedIndices,
-        Set<String> nodesWithBlockedIndices,
+        Set<String> redDataNodes,
+        Set<String> yellowDataNodes,
         Set<String> nodesReportingRed,
         Set<String> nodesReportingYellow,
         ClusterState clusterState
     ) {
-        Set<String> allUnhealthyNodes = (Stream.concat(
-            Stream.concat(nodesWithBlockedIndices.stream(), nodesReportingRed.stream()),
-            nodesReportingYellow.stream()
-        )).collect(Collectors.toSet());
-        Set<String> allRolesOnUnhealthyNodes = getRolesOnNodes(allUnhealthyNodes, clusterState).stream()
-            .map(DiscoveryNodeRole::roleName)
-            .collect(Collectors.toSet());
-        final String symptom;
-        if (clusterHasBlockedIndex && allUnhealthyNodes.isEmpty()) {
-            // In this case the disk issue has been resolved but the index block has not been automatically removed yet:
+        Set<String> unhealthyDataNodes = Sets.union(redDataNodes, yellowDataNodes);
+        Set<String> allUnhealthyNodes = Sets.union(nodesReportingRed, nodesReportingYellow);
+
+        String symptom;
+        if (blockedIndices.isEmpty() == false) {
             symptom = String.format(
                 Locale.ROOT,
-                "%d %s blocked and cannot be updated but 0 nodes are currently out of space.",
+                "%d %s not allowed to be updated because ",
                 blockedIndices.size(),
                 blockedIndices.size() == 1 ? "index is" : "indices are"
             );
+            if (unhealthyDataNodes.isEmpty()) {
+                // In this case the disk issue has been resolved but the index block has not been removed yet or the
+                // cluster is still moving shards away from data nodes that are over the high watermark.
+                symptom += ("the cluster was running out of disk space. The cluster is recovering and you should be able to update them "
```

Comment:
The "updated" repetition read a bit strange to me

```suggestion
                symptom += ("the cluster was running out of disk space. The cluster is recovering and ingest capabilities should be restored "
```